cli/users_test.go

```diff
@@ -0,0 +1,1124 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package cli_test
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/absmach/magistrala/cli"
+	// "github.com/absmach/magistrala/internal/testsutil".
+	mgclients "github.com/absmach/magistrala/pkg/clients"
+	"github.com/absmach/magistrala/pkg/errors"
+	svcerr "github.com/absmach/magistrala/pkg/errors/service"
+	mgsdk "github.com/absmach/magistrala/pkg/sdk/go"
+	sdkmocks "github.com/absmach/magistrala/pkg/sdk/mocks"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+var user = mgsdk.User{
+	Name: "testuser",
+	Credentials: mgsdk.Credentials{
+		Secret:   "testpassword",
+		Identity: "identity@example.com",
+	},
+	Status: mgclients.EnabledStatus.String(),
+}
+
+func TestCreateUsersCmd(t *testing.T) {
+	sdkMock := new(sdkmocks.SDK)
+	cli.SetSDK(sdkMock)
+	createCommand := "create"
+	usersCmd := cli.NewUsersCmd()
+	rootCmd := setFlags(usersCmd)
+
+	cases := []struct {
+		desc          string
+		args          []string
+		sdkerr        errors.SDKError
+		errLogMessage string
+		user          mgsdk.User
+		logType       outputLog
+	}{
+		{
+			desc: "create user successfully with token",
+			args: []string{
+				createCommand,
+				"john doe",
+				"john.doe@example.com",
+				"12345678",
+				"valid",
+			},
+			user:    user,
+			logType: entityLog,
+		},
+		{
+			desc: "create user successfully without token",
+			args: []string{
+				createCommand,
+				"john doe",
+				"john.doe@example.com",
+				"12345678",
+			},
+			user:    user,
+			logType: entityLog,
+		},
```

Comment:
Wrong args are passed.
```suggestion
			desc: "create user successfully with token",
			args: []string{
				createCommand,
				user.Name,
				user.Credentials.Identity,
				user.Credentials.Secret,
				token,
			},
			user:    user,
			logType: entityLog,
		},
```